chapters/memory.txt

```diff
@@ -3096,18 +3096,175 @@ ifdef::VK_KHR_device_group[]
+    <<features-memoryMapRangePlaced, pname:memoryMapRangePlaced>> feature
+    is enabled, pname:offset must: be aligned to an integer multiple of
+    sname:VkPhysicalDeviceMapMemoryPlacedPropertiesEXT::pname:minPlacedMemoryMapAlignment
+  * If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in pname:flags and the
+    <<features-memoryMapRangePlaced, pname:memoryMapRangePlaced>> feature
+    is enabled, pname:size must: be ename:VK_WHOLE_SIZE or be aligned to an
+    integer multiple of
+    sname:VkPhysicalDeviceMapMemoryPlacedPropertiesEXT::pname:minPlacedMemoryMapAlignment
+ifdef::VK_EXT_external_memory_host[]
+  * If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in pname:flags, the memory
+    object must: not have been imported from a handle type of
+    ename:VK_EXTERNAL_MEMORY_HANDLE_TYPE_HOST_ALLOCATION_BIT_EXT or
+    ename:VK_EXTERNAL_MEMORY_HANDLE_TYPE_HOST_MAPPED_FOREIGN_MEMORY_BIT_EXT
+endif::VK_EXT_external_memory_host[]
+endif::VK_EXT_map_memory_placed[]
+****
+
+include::{generated}/validity/structs/VkMemoryMapInfoEXT.txt[]
+--
+endif::VK_EXT_map_memory2[]
+
+ifdef::VK_EXT_map_memory_placed[]
+[open,refpage='VkMemoryMapPlacedInfoEXT',desc='Structure containing memory map placement parameters',type='structs']
+--
+If ename:VK_MEMORY_MAP_PLACED_BIT_EXT is set in
+sname:VkMemoryMapInfoEXT::pname:flags and the pname:pNext chain of
+slink:VkMemoryMapInfoEXT includes a sname:VkMemoryMapPlacedInfoEXT
+structure, then that structure specifies the placement address of the
+memory map.
+The implementation will attempt to place the memory map at
+that address.
+If successful, pname:ppData will be set to the same value as
+sname:VkMemoryMapPlacedInfoEXT::pname:pPlacedAddress and
+fname:vkMapMemory2EXT will return ename:VK_SUCCESS.
+If it cannot place the map at the requested address for any
+reason, the memory object is left unmapped and fname:vkMapMemory2EXT will
+return ename:VK_ERROR_MEMORY_MAP_FAILED.
+
+[NOTE]
+.Note
+====
+In order for a placed map to succeed, the client may: first have to reserve
+the virtual address range via a platform-specific mechanism.
+====
+
+The sname:VkMemoryMapPlacedInfoEXT structure is defined as:
+
+include::{generated}/api/structs/VkMemoryMapPlacedInfoEXT.txt[]
+
+  * pname:sType is the type of this structure.
+  * pname:pNext is `NULL` or a pointer to a structure extending this
+    structure.
+  * pname:pPlacedAddress is the virtual address at which to place the
+    address.  If sname:VkMemoryMapInfoEXT::pname:flags does not contain
+    ename:VK_MEMORY_MAP_PLACED_BIT_EXT, this value is ignored.
+
+.Valid Usage
+****
+  * If sname:VkMemoryMapInfoEXT::pname:flags contains
+    ename:VK_MEMORY_MAP_PLACED_BIT_EXT, pname:pPlacedAddress must: not be
+    `NULL`
+  * If sname:VkMemoryMapInfoEXT::pname:flags contains
+    ename:VK_MEMORY_MAP_PLACED_BIT_EXT, pname:pPlacedAddress must: be
+    aligned to an integer multiple of
+    sname:VkPhysicalDeviceMapMemoryPlacedPropertiesEXT::pname:minPlacedMemoryMapAlignment
```

Comment:
Both of these VUs should move to the parent structure (VkMemoryMapInfoEXT) as they reference a value in the parent structure.

Reply:
How about I drop the mention of flags and just say it has to be aligned.  0 is trivially aligned so it should be fine in 99% of the cases where the struct is unused.

Reply:
I dropped the mention of flags in the alignment one.  The other is annoying.  It really is a VU on `pPlacedAddress` not the flags.  One option would be to toss the flag and just make it based on `VkMemoryMapPlacedInfoEXT` being present in the `pNext` chain and `pPlacedAddress != NULL`.  I kind-of do like the flag, though.

Reply:
I'm fine with either resolution, though dropping the flag would make the API simpler, and make it more obvious that you can't do it with the vanilla map memory command.